src/Tribe/Views/V2/Theme_Compatibility.php

```diff
@@ -25,74 +30,94 @@ class Theme_Compatibility {
 		'twentynineteen',
 	];
 
-
 	/**
-	 * Register
+	 * Checks if theme needs a compatibility fix.
 	 *
 	 * @since  4.9.3
-	 *
-	 *
-	 * @return void
+   *
+	 * @return boolean
 	 */
-	public function register() {}
+	public function is_compatibility_required() {
+		$template = get_template();
+		$stylesheet = get_stylesheet();
+
+		// Prevents empty stylesheet or template
+		if ( empty( $template ) || empty( $stylesheet ) ) {
+			return false;
+		}
+
+		if ( in_array( $template, $this->get_registered_themes() ) ) {
+			return true;
+		}
+
+		return false;
+	}
 
 	/**
-	 * Checks if theme needs a compatibility fix.
-	 *
-	 * @param string $theme Name of template from WP_Theme->Template, defaults to current active template.
+	 * Add the theme to the body class.
 	 *
 	 * @since 4.9.3
 	 *
-	 * @return mixed
+	 * @param  array $classes Classes that are been passed to the body.
+	 *
+	 * @return array $classes
 	 */
-	public function needs_compatibility_fix( $theme = null ) {
-		// Defaults to current active theme
-		if ( $theme === null ) {
-			$theme = get_stylesheet();
+	public function filter_add_body_classes( array $classes ) {
+		if ( ! tribe( Template_Bootstrap::class )->should_load() ) {
+			return $classes;
 		}
 
-		/**
-		 * Allows to filter the theme list with compatibility fixes.
-		 *
-		 * @since  4.9.3
-		 *
-		 * @param  array  $themes_with_compatibility_fixes A list of themes we provide compatibility for.
-		 */
-		$theme_compatibility_list = apply_filters( 'tribe_events_views_v2_themes_compatibility_fixes', $this->themes_with_compatibility_fixes );
```

Comment:
We would keep this filter. That way we could add an extension to give compatibility with whatever theme we want.

Reply:
We moved to use `tribe_events_views_v2_theme_compatibility_registered` which happens on a get method.

Reply:
awesome 🎉